api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/rtt/ReturnedAdnITTDifferenceDateCalculation.java

```diff
@@ -0,0 +1,330 @@
+            + "								WHERE  pa.voided = 0 "
+            + "                                AND enc.voided = 0 "
+            + "                                AND enc.encounter_type IN ( ${adultoSeguimentoEncounterType},${aRVPediatriaSeguimentoEncounterType} ) "
+            + "                                AND enc.location_id = :location "
+            + "                                AND enc.encounter_datetime <= :previousPeriod "
+            + "							GROUP  BY pa.patient_id) ficha "
+            + "                        INNER JOIN encounter e on "
+            + "                            e.patient_id = ficha.patient_id and "
+            + "                            e.encounter_datetime = ficha.encounter_datetime and "
+            + "                            e.encounter_type IN (${adultoSeguimentoEncounterType},${aRVPediatriaSeguimentoEncounterType}) and "
+            + "                            e.location_id = :location and "
+            + "                            e.voided = 0 "
+            + "                        INNER JOIN obs o on "
+            + "                            o.encounter_id = e.encounter_id and "
+            + "                            o.concept_id = ${returnVisitDateConcept} and "
+            + "                            o.voided = 0 "
+            + "                        UNION "
+            + "                        SELECT pa.patient_id, "
+            + "                            Date_add(Max(obs.value_datetime), interval 30 day) value_datetime "
+            + "                        FROM   patient pa "
+            + "                            inner join encounter enc "
+            + "                                ON enc.patient_id = pa.patient_id "
+            + "                            inner join obs obs "
+            + "                                ON obs.encounter_id = enc.encounter_id "
+            + "                        WHERE  pa.voided = 0 "
+            + "                            AND enc.voided = 0 "
+            + "                            AND obs.voided = 0 "
+            + "                            AND obs.concept_id = ${artDatePickup} "
+            + "                            AND obs.value_datetime IS NOT NULL "
+            + "                            AND enc.encounter_type = ${msterCardDrugPickupEncounterType}  "
+            + "                            AND enc.location_id = :location "
+            + "                            AND obs.value_datetime <= :previousPeriod "
+            + "                       GROUP  BY pa.patient_id "
+            + "                   ) most_recent "
+            + "               GROUP BY most_recent.patient_id "
+            + "               HAVING final_encounter_date < :previousPeriod ";
+
+    StringSubstitutor stringSubstitutor1 = new StringSubstitutor(map1);
+    String replacedQeury1 = stringSubstitutor1.replace(query1);
+
+    String finalQuery = replacedQeury1;
```

Comment:
You can just initialize this variable with stringSubstitutor1.replace(query1); directly, to remove variable redundacny

Reply:
done